internal/ark/backups_repository.go

```diff
@@ -49,6 +49,18 @@ func (r *BackupsRepository) Find() (backups []*ClusterBackupsModel, err error) {
 	return
 }
 
+// Find returns ClusterBackupsModel instances
+func (r *BackupsRepository) FindByClusterID(clusterID uint) (backups []*ClusterBackupsModel, err error) {
+	query := ClusterBackupsModel{
+		OrganizationID: r.org.ID,
+		ClusterID:      clusterID,
+	}
+
+	err = r.db.Where(&query).Preload("Bucket").Preload("Bucket.Deployment").Preload("Organization").Find(&backups).Error
+
+	return
```

Comment:
**Optional**: IMO explicit error check or direct return is easier to read and less error prone for future extensions here.

```suggestion
	return r.db.
	    Where(&query).
	    Preload("Bucket").
	    Preload("Bucket.Deployment").
	    Preload("Organization").
	    Find(&backups).
	    Error
```